koa_crud/test/integration/use-cases/promos/promos.spec.ts

```diff
@@ -0,0 +1,430 @@
+          'Description must be provided.',
+        );
+      });
+
+      it('should throw an error for empty minimum balance given template is deposit', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: null,
+        };
+
+        await expect(insertPromo(data)).to.eventually.rejectedWith(
+          'Minimum balance must be provided.',
+        );
+      });
+
+      it('should throw an error for empty promo fields given template is sign up', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.SignUp,
+          title: chance.word(),
+          description: chance.sentence(),
+          requiredMemberFields: [],
+        };
+
+        await expect(insertPromo(data)).to.eventually.rejectedWith(
+          'Members fields must be provided.',
+        );
+      });
+
+      it('should throw an error for invalid promo fields given template is sign up', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.SignUp,
+          title: chance.word(),
+          description: chance.sentence(),
+          requiredMemberFields: ['Test'],
+        };
+
+        await expect(insertPromo(data)).to.eventually.rejectedWith(
+          'Test is an invalid field.',
+        );
+      });
+
+      it('should throw an error for invalid status', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.SignUp,
+          title: chance.word(),
+          description: chance.sentence(),
+          requiredMemberFields: [MemberFields.REAL_NAME],
+          status: 'Test',
+        };
+
+        await expect(insertPromo(data)).to.eventually.rejectedWith(
+          'Invalid status.',
+        );
+      });
+    });
+
+    context('Given correct values', () => {
+      it('should insert promo', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+        };
+
+        const result = await insertPromo(data);
+
+        expect(result).to.exist;
+        expect(result).to.be.true;
+      });
+    });
+  });
+
+  describe('List Promos', () => {
+    it('should return all promos', async () => {
+      const result = await listPromos();
+
+      expect(result).to.exist;
+      expect(result).to.be.an('array');
+      expect(result).to.have.length.greaterThan(0);
+    });
+
+    it('should return one promo', async () => {
+      const promo = await listPromos();
+
+      const lastPromoId = R.compose(R.prop('id'), R.last)(promo);
+
+      const result = await selectPromo(lastPromoId);
+
+      expect(result).to.exist;
+      expect(result).to.be.an('object');
```

Comment:
Can you add another assertion where it compares the `result.id` and the `lastPromoId`?
I think that would also be helpful

Reply:
Nice :heavy_check_mark: